segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/StorageWriter.java

```diff
@@ -303,6 +303,19 @@ private CompletableFuture<Void> processMetadataOperation(MetadataOperation op) {
     }
 
     private CompletableFuture<Void> processSegmentOperation(SegmentOperation op) {
+        val segmentMetadata = this.dataSource.getStreamSegmentMetadata(op.getStreamSegmentId());
+        if ( segmentMetadata == null ) {
```

Comment:
Can you elaborate a little bit on the type of recovery process you are doing to get to this point? It is quite strange that an operation that has passed through the `OperationProcessor` has no metadata, basically because the `OperationProcessor` is intended to create and/or update the metadata of segments that it processes. 

What is more important: what are the consequences of not processing this operation? Should the log be a warning? Should it be better to understand if we need to add any further "checks" at the `OperationProcessor` level to prevent an inconsistent state rather than just skipping unexpected operations at this point of the write path? cc/ @tkaitchuck

Reply:
Also, please remove spaces after `( ` and before ` )`.

Reply:
This PR had the issue creation pending which is now updated in the description of the PR which elaborates the issue more.